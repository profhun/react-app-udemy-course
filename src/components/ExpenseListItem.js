import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <li>
        <h4>{description}</h4>
        <div>{amount}</div>
        <div>{createdAt}</div>
        <Link to={`/edit/${id}`}>edit</Link>
    </li>
);

export default ExpenseListItem;