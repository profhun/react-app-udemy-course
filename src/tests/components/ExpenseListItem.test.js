import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'

test('render/component ExpenseListItem', () => {

    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper.find('h4').text()).toBe(expenses[0].description);
    expect(wrapper).toMatchSnapshot();
});