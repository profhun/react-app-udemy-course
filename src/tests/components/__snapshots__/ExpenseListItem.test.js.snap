// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render/component ExpenseListItem 1`] = `
<li>
  <h4>
    zokni a l√°bon
  </h4>
  <div>
    400
  </div>
  <div>
    -864000000
  </div>
  <Link
    replace={false}
    to="/edit/zokni"
  >
    edit
  </Link>
</li>
`;
